src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketTagClientImpl.java

```diff
@@ -61,24 +61,20 @@ public Stream<BitbucketTag> getRemoteTags() {
                 .flatMap(Collection::stream);
     }
 
-    public BitbucketTag getRemoteTag(String tagName) {
-        HttpUrl.Builder urlBuilder = null;
-        try {
-            urlBuilder = bitbucketRequestExecutor.getCoreRestPath().newBuilder()
+    public BitbucketSingleTag getRemoteTag(String tagName) {
```

Comment:
As mentioned before, if we're actually fetching the commit associated with the tag, lets just return a `BitbucketCommit` here instead, and perhaps have the method name reflect that too.

Reply:
i've opted to revert back to using a BitbucketTag as thats what we've used previously. We just construct it from the commit that comes back.